### PR TITLE
Add sub organization  (Task #199447)

### PR DESCRIPTION
**Pull request by poojakarma**
Steps for Implementation

- Add-Sub-ORG button on UI
- create one component for sub-org
- create form for add sub-org
- create function for add-sub-org in ts file
- create function in service to call API for sub-org
- User language constant 
- Warning msg 

video : https://drive.google.com/file/d/1QN71jvbkqzUzJu5RPMVLs4KKxaNPPtTb/view?usp=drive_link

### CODE REVIEW

File: src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  styleUrls: ['./add-sub-org.component.scss']
+})
+export class AddSubOrgComponent {
+  addSubOrg: FormGroup;
+  submitted: boolean = false;
+  messages!: Message[];
+  organizations: any[] = [];
+  selectedOrg: any[] = []
+
+  constructor(public formBuilder: FormBuilder, public ref: DynamicDialogRef, private addSUBOrg: OrganizationListService, private orgList: UserService, private i18nextPipe: I18NextPipe) {
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts]: ref => dynamicDialogRef

File: src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import { Message } from 'primeng/api';
+import { UserService } from 'src/app/sb-admin/service/user.service';
+
+@Component({
+  selector: 'app-add-sub-org',
+  templateUrl: './add-sub-org.component.html',
+  styleUrls: ['./add-sub-org.component.scss']
+})
+export class AddSubOrgComponent {
+  addSubOrg: FormGroup;
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts]: Is this for formGroup or OrganizationListService?

File: src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts (hunk 1 above)
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts]: Use proper naming convention
addSUBOrg => organizationListService

File: src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    const updatedFormValue = {
+      ...this.addSubOrg.value,
+      channel: channel,
+      rootOrgId: rootOrgId
+    }
+    delete(updatedFormValue.rootOrgName)
+    const body = {
+      "request": updatedFormValue
+    }
+    this.addSUBOrg.addSubOrg(body).subscribe((response) => {
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts]: addSUBOrg => organizationListService

File: src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+      (error: any) => {
+        this.messages = [
+          { severity: 'error', summary: this.i18nextPipe.transform('ADD_SUB_ORGANIZATION_ALREADY_EXIT') }
+        ]
+      }
+    )
+  }
+
+  cancel() {
+    this.ref.close();
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts]: ref => dynamicDialogRef

File: src/app/sb-admin/service/organization-list.service.ts [hunk 5]
```diff
@@ -26,4 +26,15 @@ export class OrganizationListService {
     })
  return this.http.post(config.URLS.ORGANIZATION_CREATE_URL,body,{headers: header})
 }
+
+addSubOrg(body : any): Observable<any>{
+  console.log(body)
```
Comment (gouravmore) [inline src/app/sb-admin/service/organization-list.service.ts]: remove console log

File: src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+      ...this.addSubOrg.value,
+      channel: channel,
+      rootOrgId: rootOrgId
+    }
+    delete(updatedFormValue.rootOrgName)
+    const body = {
+      "request": updatedFormValue
+    }
+    this.addSUBOrg.addSubOrg(body).subscribe((response) => {
+      console.log(response);
```
Comment (gouravmore) [inline src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts]: remove console log

**Review by gouravmore — CHANGES_REQUESTED** [review]

File: src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts (hunk 2 above)
Comment (poojakarma) [reply src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts]: sir for form Group

File: src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts (hunk 3 above)
Comment (poojakarma) [reply src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts]: Resolved

File: src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts (hunk 4 above)
Comment (poojakarma) [reply src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts]: Resolved

File: src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts (hunk 6 above)
Comment (poojakarma) [reply src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.ts]: Removed

File: src/app/sb-admin/service/organization-list.service.ts (hunk 5 above)
Comment (poojakarma) [reply src/app/sb-admin/service/organization-list.service.ts]: Removed

File: src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.scss [hunk 7]
```diff
@@ -0,0 +1,12 @@
+input.ng-touched.ng-invalid{
+    border: 1px solid red;
+}
+textarea.ng-touched.ng-invalid{
+    border: 1px solid red;
+}
+small{
```
Comment (manojLondhe) [inline src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.scss]: small or red - can we have logical names for classes?

File: src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.scss [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+    color: red;
+}
+.p-button-container{
```
Comment (manojLondhe) [inline src/app/sb-admin/components/pages/sb-organization/add-sub-org/add-sub-org.component.scss]: can we have logical names for classes?

File: src/app/sb-admin/service/organization-list.service.ts [hunk 9]
```diff
@@ -26,4 +26,14 @@ export class OrganizationListService {
     })
  return this.http.post(config.URLS.ORGANIZATION_CREATE_URL,body,{headers: header})
 }
+
+addSubOrg(body : any): Observable<any>{
```
Comment (manojLondhe) [inline src/app/sb-admin/service/organization-list.service.ts]: format code using vscode prettier extension

File: src/app/sb-admin/components/pages/sb-organization/sb-organization.component.ts [hunk 10]
```diff
@@ -62,11 +73,23 @@ export class SbOrganizationComponent implements OnDestroy {
...
   }
+
+  addSubOrg() {
+    this.ref=this.dialogService.open(AddSubOrgComponent, this.addSubOrgDialog);
+    this.ref.onClose.subscribe((newSubOrgData: any) => {
+      if (newSubOrgData) {
+        this.messageService.add({ severity: 'success', summary: this.i18nextPipe.transform('ADD_SUB_ORGANIZATION_SUCCESSFULLY') })
+      }
+    })
+
```
Comment (manojLondhe) [inline src/app/sb-admin/components/pages/sb-organization/sb-organization.component.ts]: format code using vscode prettier extension

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: src/app/sb-admin/service/organization-list.service.ts (hunk 9 above)
Comment (poojakarma) [reply src/app/sb-admin/service/organization-list.service.ts]: sir I am checking this extension

**Pull request merged**